modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/NashornAdaptiveScriptInitializerTestCase.java

```diff
@@ -18,38 +18,138 @@
 
-    private ServerConfigurationManager scm;
+    private ServerConfigurationManager serverConfigurationManager;
     private File defaultConfigFile;
 
+    private int javaVersion;
+    private static String enableInfo = "Adaptive authentication successfully enabled.";
+    private static String disableInfo = "Adaptive authentication successfully disabled.";
+
     @BeforeTest(alwaysRun = true)
-    public void initScriptEngineConfig() throws Exception {
+    public void testInit() throws Exception {
 
         super.init();
+        serverConfigurationManager = new ServerConfigurationManager(isServer);
         String carbonHome = CarbonUtils.getCarbonHome();
         defaultConfigFile = getDeploymentTomlFile(carbonHome);
+
+        javaVersion = Utils.getJavaVersion();
+        String identityNewResourceFileName = "nashorn_script_engine_config.toml";
+
+        if (javaVersion >= 15) {
+            // Download OpenJDK Nashorn only if the JDK version is Higher or Equal to 15.
+            runAdaptiveAuthenticationDependencyScript(false);
+            identityNewResourceFileName = "openjdknashorn_script_engine_config.toml";
+        }
+
         File scriptEngineConfigFile = new File(
                 getISResourceLocation() + File.separator + "scriptEngine" + File.separator +
-                        "nashorn_script_engine_config.toml");
-        scm = new ServerConfigurationManager(isServer);
-        scm.applyConfiguration(scriptEngineConfigFile, defaultConfigFile, true, true);
+                        identityNewResourceFileName);
+        serverConfigurationManager.applyConfiguration(scriptEngineConfigFile, defaultConfigFile, true, true);
+    }
+
+    private void runAdaptiveAuthenticationDependencyScript(boolean disable) {
+
+        ServerLogReader inputStreamHandler;
+        ServerLogReader errorStreamHandler;
+        String targetFolder = System.getProperty("carbon.home");
+        String scriptFolder = getTestArtifactLocation() + File.separator;
+        Process tempProcess = null;
+        File scriptFile = new File(scriptFolder);
+        Runtime runtime = Runtime.getRuntime();
+
+        try {
+            if (System.getProperty("os.name").toLowerCase().contains("windows")) {
+                log.info("Operating System is Windows. Executing batch script");
+                if (disable) {
+                    // TODO https://github.com/wso2/product-is/issues/14301
+                    restartServer();
+                    tempProcess = runtime.exec(
+                            new String[] { "cmd", "/c", "adaptive.bat", targetFolder, "DISABLE" }, null, scriptFile);
+                } else {
+                    tempProcess = runtime.exec(
+                            new String[] { "cmd", "/c", "adaptive.bat", targetFolder }, null, scriptFile);
+                }
+                errorStreamHandler = new ServerLogReader("errorStream",
+                        tempProcess.getErrorStream());
```

Comment:
This line seems to be able to merge. Same with the next line as well.

Reply:
Resolved 58d1ce72a65fbb035f1639420836bb04e214602b